q2cli/builtin/tools.py

```diff
@@ -907,3 +908,249 @@ def cache_status(cache):
     success = "Status of the cache at the path '%s':\n\n%s" % \
         (cache, output)
     click.echo(CONFIG.cfg_style('success', success))
+
+
+@tools.command(name='replay-provenance', cls=ToolCommand)
+@click.option('--in-fp', required=True,
+              help='filepath to a QIIME 2 Archive or directory of Archives')
+@click.option('--recurse/--no-recurse',
+              default=False,
+              show_default=True,
+              help=('if in-fp is a directory, will also search sub-directories'
+                    ' when finding .qza/.qzv files to parse'))
+@click.option('--usage-driver',
+              default='cli',
+              show_default=True,
+              help='the target interface for your replay script',
+              type=click.Choice(['python3', 'cli'], case_sensitive=False))
+@click.option('--validate-checksums/--no-validate-checksums',
+              default=True,
+              show_default=True,
+              help='check that replayed archives are intact and uncorrupted')
+@click.option('--parse-metadata/--no-parse-metadata',
+              default=True,
+              show_default=True,
+              help=('parse the original metadata captured in provenance '
+                    'for review or replay'))
+@click.option('--use-recorded-metadata/--no-use-recorded-metadata',
+              default=False,
+              show_default=True,
+              help='re-use the original metadata captured in provenance')
+@click.option('--suppress-header/--no-suppress-header',
+              default=False,
+              show_default=True,
+              help='do not write header/footer blocks in the output script')
+@click.option('--verbose/--no-verbose',
+              default=True,
+              show_default=True,
+              help='print status messages to stdout while processing')
+@click.option('--dump-recorded-metadata/--no-dump-recorded-metadata',
+              default=True,
+              show_default=True,
+              help='write the original metadata captured in provenance to '
+                   'disk in the --metadata-out-fp directory')
+@click.option('--metadata-out-fp',
+              default='',
+              show_default=True,
+              help=('the directory where captured study metadata '
+                    'should be written if --dump-recorded-metadata. This '
+                    'often produces many outputs, so a dedicated directory '
+                    'should generally be used. Creates the directory if it '
+                    'does not already exist. By default, metadata is written '
+                    'to `${PWD}/recorded_metadata/`'))
+@click.option('--out-fp',
```

Comment:
We could make this a [click.Path](https://github.com/qiime2/q2cli/blob/6c91089e0fad52b3558586f93cdb8f01b0379134/q2cli/builtin/dev.py#L78) exists=False, writable=True